simulation_rate: 100.0 # Hz
clock_rate:      400.0 # Hz, should be the same rate or higher than the simulation rate
realtime_factor: 1.0 # [-]

# when true, the simulation time will get initialized from wall time
sim_time_from_wall_time: true

g: 9.81 # [ms^-2]

iterate_without_input: true

input_timeout: 1.0 # [s]

subscribers:

  # used for feedforward
  tracker_cmd:
    enabled: true

  position_cmd:
    enabled: true

  velocity_hdg_cmd:
    enabled: true

  velocity_hdg_rate_cmd:
    enabled: true

  acceleration_hdg_cmd:
    enabled: true

  acceleration_hdg_rate_cmd:
    enabled: true

  attitude_cmd:
    enabled: true

  attitude_rate_cmd:
    enabled: true

  control_group_cmd:
    enabled: true

  actuators_group_cmd:
    enabled: true

publishers:

  odometry:
    enabled: true

  rangefinder:
    enabled: true

  imu:
    enabled: true

frames:
  world:
    name: "simulator_origin"
    prefix_with_uav_name: false
  fcu:
    name: "fcu"
    publish_tf: false
  rangefinder:
    name: "garmin"
    publish_tf: true

collisions:
  enabled: true
  crash: true

  # if crash==false, then the collisions are elastic
  rebounce: 100.0

# randomization of the initial conditions around the desired spawn point
randomization:

  enabled: false

  bounds:
    x: 15.0
    y: 15.0
    z: 15.0

ground:
  enabled: true
  z: 0.0 # [m]

# * when enabled, this will place a temporary patch of ground just under the spawn location
# * this patch will disappear after the total_thrust > 0.9*hover_thrust
individual_takeoff_platform:
  enabled: false
